packages/plugin-core/src/commands/InsertNoteIndexCommand.ts

```diff
@@ -0,0 +1,78 @@
+import { DENDRON_COMMANDS } from "../constants";
+import { BasicCommand } from "./base";
+import { VSCodeUtils } from "../utils";
+import { DNodeUtils, NoteProps } from "@dendronhq/common-all";
+import { getEngine } from "../workspace";
+import { window } from "vscode";
+import _ from "lodash";
+
+type CommandOpts = {
+  /*
+   * optional flag that will wrap the index block with markers.
+   */
+  marker?: boolean;
+};
+
+type CommandOutput = CommandOpts;
+
+export class InsertNoteIndexCommand extends BasicCommand<
+  CommandOpts,
+  CommandOutput
+> {
+  key = DENDRON_COMMANDS.INSERT_NOTE_INDEX.key;
+
+  // TODO: make this into a util once the cli version is implemented.
+  // NOTE: the marker flag is not exposed to the plugin yet.
+  genNoteIndex(notes: NoteProps[], opts: {
+    marker?: boolean
```

Comment:
Should we make this default to true?

Reply:
My reasoning to default it to be false is because if the user is triggering the command themselves in a note, they are adding the index block as the _content_ of the note. This means the plugin should not look at the note and try to update the index so I felt the markers would be irrelevant here.

I'm of the opinion that the markers should only be in the notes that we are going to implement as best practice, and for truly autogenerated ones like the ones that would be added in bulk via export pod or cli. If the user wants to have the markers added when using this command themselves, it should be passed in as an argument instead of it being the default.

Does adding a new flag in `dendron.yml` sound like a good middle ground?